### DISCUSSION TITLE
Import model as space

### DISCUSSION THREAD

**Discussion by AB-Athene** [fumitoh/modelx#161] [Ideas]
In case of broader adoption of `modelx` use by the organization, naturally different individuals/teams start to own different models (modelling logic, data sets, ...). Examples are:
- one team models endowment liability product, another - term
- one team models in-force business, another - new business
- one team models assets, another - liabilities, ...

In this situation ideal process would be
1. Each team produces `modelx` model object including all the data, parameters and formulas.
2. There is a way to merge all these models into one and build overarching logic on top.

To facilitate it, it seems relatively straightforward extension of the `modelx` framework to allow import of `model ` as `space`.

**Comment by fumitoh** [comment]
This is not documented, but you can actually copy a space to another model, by `UserSpace.copy`. This usage is experimental, and it probably causes an issue when the elements of the copied space have `refs` associated with files, but it should work in simple cases as below.

```python
import modelx as mx

# Create Model1:
#
# Model1---A---foo

m1 = mx.new_model()
m1.new_space('A')

@mx.defcells
def foo():
    return 1

# Create Model2:
#
# Model2---B---bar

m2 = mx.new_model()    
m2.new_space('B')

@mx.defcells
def bar():
    return 2

# Copy A to Model2:
#
# Model2---B---bar
#        +-A---foo
    
m1.A.copy(m2)

```

**Reply by AB-Athene** [reply]
File refs is one complication. Another is interdependency between spaces in the source model.

**Reply by fumitoh** [reply]
Yes. File refs can be fixed, but how to treat space interdependency upon copy requires some thoughts.

**Reply by AB-Athene** [reply]
What do you think about ability to import model as a space with model spaces becoming child spaces of the created space?

**Reply by fumitoh** [reply]
That’s a good idea. Let me think about how to implement it.

**Reply by declann** [reply]
Sounds interesting if this can/will be done
